pkg/kubectl/cmd/set/set_port.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/api"
```

Comment:
We are not introducing new deps on k8s.io/kubernetes.  How can this PR be done without this?  Can we use the versioned objects.

Reply:
we want to do this anyway, see the TODO in https://github.com/kubernetes/kubernetes/pull/53158/files#diff-466503ed5c81843c5ba705bc4c6fd4dc

> // TODO: switch UpdatePodSpecForObject to work on v1.PodSpec, use info.VersionedObject, and avoid conversion completely

Reply:
Why a TODO?  Can we do this now before this PR goes in?  We are actively trying to remove these dependencies from kubectl.

Reply:
agree, was saying we could use this as an opportunity to resolve the TODO

Reply:
oh perfect :)

Reply:
there is  a issue created by wangshiyang. will fix it priority then fix this PR

Reply:
open PR #54554 fix this